kingpin/actors/base.py

```diff
@@ -597,21 +602,30 @@ def _gather_methods(self):
         """
         self.setters = {}
         self.getters = {}
-        for option in self.all_options:
+        self.comparers = {}
+        for option in self._ensurable_options:
             setter = '_set_%s' % option
             getter = '_get_%s' % option
+            comparer = '_compare_%s' % option
 
             if not self._is_method(getter) or not self._is_method(setter):
-                if option in self.unmanaged_options:
-                    continue
-
                 raise exceptions.UnrecoverableActorFailure(
                     'Invalid Actor Code Detected in %s: '
                     'Unable to find required methods: %s, %s'
                     % (self.__class__.__name__, setter, getter))
 
+            if not self._is_method(comparer):
```

Comment:
Whoa... seems convoluted. Why not go ahead and define this method and let the subclasses overwrite it? Either way need a fat docstring